src/components/side-bar/with-anchors.js

```diff
@@ -0,0 +1,113 @@
+import PropTypes from 'prop-types'
+import React from 'react'
+import get from 'lodash/get'
+import hoistStatics from 'hoist-non-react-statics'
+
+const _ = {
+  get,
+}
+
+/**
+ * Get react component display name
+ *
+ * @param {object} DecoratedComponent
+ * @returns {string} display name of react component
+ */
+function getDisplayName(DecoratedComponent) {
+  return (
+    DecoratedComponent.displayName || DecoratedComponent.name || 'Component'
+  )
+}
+
+function AnchorWrapper({ children }) {
+  return React.Children.only(children)
+}
+
+AnchorWrapper.propTypes = {
+  anchorId: PropTypes.string.isRequired,
+  anchorLabel: PropTypes.string.isRequired,
+  children: PropTypes.element.isRequired,
+}
+
+/**
+ * `withAnchors`
+ * returns a High Order Component setting the anchors onto `DecoratedComponent`.
+ * It passes props `anchors` to the `DecoratedComponent` which subsequently adds the side bar functionalities to its children.
+ * Note that it should be utilized along with `AnchorWrapper`.
+ *
+ * For example,
+ * ```
+ *
+ * import IndexPageSideBar from './index-page-side-bar'
+ * import SideBarHOC from '@twreporter/react-components/lib/side-bar'
+ *
+ * const SideBar = withAnchors(SideBarHOC(IndexPageSideBar))
+ *
+ * const ContentWithAnchors = () => {
+ *  return (
+ *    <Sidebar>
+ *      <AnchorWrapper anchorId="section-1" anchorLabel="Section One">
+ *        <SectionOne />
+ *      </AnchorWrapper>
+ *      <AnchorWrapper anchorId="section-2" anchorLabel="Section Two">
+ *        <SectionTwo />
+ *      </AnchorWrapper>
+ *      <AnchorWrapper anchorId="section-3" anchorLabel="Section Three">
+ *        <SectionThree />
+ *      </AnchorWrapper>
+ *    </Sidebar>
+ *  )
+ * }
+ *```
+ *
+ * The above codes will get `anchorId`, `anchorLabel` through <AnchorWrapper /> props and generates
+ * the anchors array which is the prop required by SibeBarHOC as shown below.
+ *
+ * ```
+ * const anchors = [{
+ *  id: 'section-1',
+ *  label: 'Section One'
+ * }, {
+ *  id: 'section-2',
+ *  label: 'Section Two'
+ * }, {
+ *  id: 'section-3',
+ *  label: 'Section Three'
+ * }]
+ * ```
+ *
+ * @param {Object} DecoratedComponent A React component wants to have side bar functionality
+ * @returns {Object} A React component with anchors props
+ */
+function withAnchors(DecoratedComponent) {
```

Comment:
These codes are similar to these in @twreporter-npm-packages/react-components/src/side-bar. Why do we choose to implement it independently rather than in the SideBarHOC?

Reply:
Firstly, I wanted to separate functions `collecting anchors` from `generating side bar`.

But I'm considering this suggestion to merge `withAnchors` into `SideBarHOC` since if it is a good practice to declare all the anchors via the way of this change, we could make `SideBarHOC` to adopt such usage by default.